docs/user-defined-functions-c#.md

```diff
@@ -0,0 +1,43 @@
+# User-Defined Functions - C#
+This documentation contains user-defined function (UDF) examples. It shows how to define UDFs and how to use UDFs with Row objects as examples.
+
+## Pre-requisites:
+Install Microsoft.Spark.Worker. When you want to execute a C# UDF, Spark needs to understand how to launch the .NET CLR to execute this UDF. Microsoft.Spark.Worker provides a collection of classes to Spark that enable this functionality. Please see more details at [how to install Microsoft.Spark.Worker](https://docs.microsoft.com/en-us/dotnet/spark/tutorials/get-started#5-install-net-for-apache-spark) and [how to deploy worker and UDF binaries](https://docs.microsoft.com/en-us/dotnet/spark/how-to-guides/deploy-worker-udf-binaries).
+
+## UDF that takes in Row objects
```

Comment:
Why are we only showing examples of UDFs with Row objects? It seems like it'd be valuable to have this document explain how to write any UDF and show examples of all (or at least more types) of UDFs? 

Or is the goal of this doc to only show Row-based UDFs (in this case, we should change the title and intro of the doc to reflect that, because right now it seems like it should explain all UDFs)?

Reply:
I think the purpose of this doc is `using UDF with Row objects` readme file. This goes with the recent PR which exposes the UDF that returns `Row` objects. I think we can add more types later. @imback82 what do you think?

Reply:
We can start with UDFs with `Row` since there are few gotchas with them, and we can expand this.

Reply:
Sounds good!